### ISSUE TITLE
 all modal data must be used during model training？

### ISSUE THREAD

**Issue by Sarah-air** [konst-int-i/healnet#18]
In this work, all modal data must be used during model training, but missing modal data can be used during inference, right?

**Comment by konst-int-i** [comment]
Hi @Sarah-air, 

The results in the paper were produced using all modalities for training and the ablation study for missing modalities then masks data during inference. 

However, you can run training with missing modalities for individual samples. While we do not do this in the paper, the architecture is designed to handle this. Whether this is effective will probably depend on your specific dataset, but we hope that this will provide an option to choose training on more samples (where some may have individual modalities missing). 

I am currently working on a camera-ready release which should come in the next few days. In there, we will provide some tutorials that will specify how to run the "missing modality training". 

Let me know if you have any questions in the meantime. 

Thanks!

**Comment by konst-int-i** [comment]
Dear @Sarah-air,

We added all the latest changes, including a tutorial on handling the missing modalities in https://github.com/konst-int-i/healnet/pull/22.

I would recommend looking at the last section of the [tutorial](https://github.com/konst-int-i/healnet/blob/main/tutorial/01_Getting_Started.ipynb) for further details. Note that the modalities should be passed in as missing explicitly, i.e.,

```python
logits = model([tab_sample, None, img_sample_3d], verbose=True)
```

```bash
Missing modalities indices: [1]
Skipping update in fusion layer 1 for missing modality 2
Skipping update in fusion layer 2 for missing modality 2
Skipping update in fusion layer 3 for missing modality 2
```

Let me know if you encounter any further issues.

Thanks!

**Comment by Sarah-air** [comment]
Thank you for your reply



发自我的iPhone


------------------ Original ------------------
From: Konstantin Hemker ***@***.***&gt;
Date: Wed,Dec 11,2024 0:16 AM
To: konst-int-i/healnet ***@***.***&gt;
Cc: Sarah ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [konst-int-i/healnet]  all modal data must be used during model training？ (Issue #18)





 
Dear @Sarah-air,
 
We added all the latest changes, including a tutorial on handling the missing modalities in #22.
 
I would recommend looking at the last section of the tutorial for further details. Note that the modalities should be passed in as missing explicitly, i.e.,
 logits = model([tab_sample, None, img_sample_3d], verbose=True)
 Missing modalities indices: [1] Skipping update in fusion layer 1 for missing modality 2 Skipping update in fusion layer 2 for missing modality 2 Skipping update in fusion layer 3 for missing modality 2
 
Let me know if you encounter any further issues.
 
Thanks!
